lib/services/network/socket_service.dart

```diff
@@ -155,39 +164,35 @@ class SocketService extends GetxService {
         _reconnectTimer = null;
         NetworkTasks.onConnect();
         notif.clearSocketError();
-        return;
+        break;
       case SocketState.disconnected:
         Logger.info("Disconnected from socket...");
         state.value = SocketState.disconnected;
-        return;
+        break;
       case SocketState.connecting:
         Logger.info("Connecting to socket...");
         state.value = SocketState.connecting;
-        return;
+        break;
       case SocketState.error:
```

Comment:
I'm not sure i liked that we changed the functionality of fetching from firebase. Especially when not everyone will have a secondary address entered.

Can we maintain the old functionality based on whether or not there is a secondary server address set? If not set, fetch new url from firebase. I also think that if both server URLs are tried before it works, it should also fetch the new url from firebase. Essentially, keep a counter of errors/failures. It's incremented on error, and reset on connection. If errors == serverAddresses.length, then also try fetching new url from firebase

Reply:
> I'm not sure i liked that we changed the functionality of fetching from firebase. Especially when not everyone will have a secondary address entered.
> 
> Can we maintain the old functionality based on whether or not there is a secondary server address set? If not set, fetch new url from firebase. I also think that if both server URLs are tried before it works, it should also fetch the new url from firebase. Essentially, keep a counter of errors/failures. It's incremented on error, and reset on connection. If errors == serverAddresses.length, then also try fetching new url from firebase



Ah yes I agree that is a better method will do that